ui/app/components/auth-jwt.js

```diff
@@ -0,0 +1,163 @@
+import Ember from 'ember';
+import { inject as service } from '@ember/service';
+import Component from './outer-html';
+import { next, later } from '@ember/runloop';
+import { task, timeout, waitForEvent } from 'ember-concurrency';
+import { computed } from '@ember/object';
+
+const WAIT_TIME = Ember.testing ? 0 : 500;
+const ERROR_WINDOW_CLOSED =
+  'The provider window was closed before authentication was complete.  Please click Sign In to try again.';
+const ERROR_MISSING_PARAMS =
+  'The callback from the provider did not supply all of the required parameters.  Please click Sign In to try again. If the problem persists, you may want to contact your administrator.';
+
```

Comment:
Hmm, not without another level of indirection or two and then it would still need to pass a string which feels like it wouldn't gain anything over this.  Currently the error message component takes an error or a model and will display when that value isn't falsey (and if isError is true in the case that it's a model). Most of the time the error comes from the server so it's dynamic. The other tricky part here is that the component itself doesn't render the error, it passes the error into an action and the parent component (`auth-form.js`) renders the error.

Reply:
Aaah cool np, don't worry then, I didn't look into it properly, so I had no idea what other impact that would have - I think this just comes from me trying to get all my copy into templates over in Consul-land.